modules/apps/adaptive-media/adaptive-media-web/src/main/resources/META-INF/resources/adaptive_media/js/EditAdaptiveMedia.js

```diff
@@ -0,0 +1,369 @@
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayCheckbox, ClayRadio, ClayRadioGroup} from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import {useFormik} from 'formik';
+import {
+	fetch,
+	normalizeFriendlyURL,
+	objectToFormData,
+	openToast,
+} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useCallback, useState} from 'react';
+
+import {HelpMessage, Input, RequiredMark} from './form/Components';
+import {alphanumeric, required, validate} from './form/validations';
+
+const scrollToTop = () => window.scrollTo({behavior: 'smooth', top: 0});
+
+const EditAdaptiveMedia = ({
+	actionUrl,
+	amImageConfigurationEntry,
+	automaticUuid,
+	configurationEntryEditable,
+	configurationEntryUuid,
+	namespace,
+	redirect,
+}) => {
+	const [automaticId, setAutomaticId] = useState(automaticUuid);
+	const [addHighResolution, setAddHighResolution] = useState(false);
+	const [errorMessage, setErrorMessage] = useState(null);
+
+	const nameId = `${namespace}name`;
+	const descriptionId = `${namespace}description`;
+	const maxWidthId = `${namespace}maxWidth`;
+	const maxHeightId = `${namespace}maxHeight`;
+	const highResolutionId = `${namespace}addHighResolution`;
+	const newUuidId = `${namespace}newUuid`;
+	const automaticRadioId = `${namespace}automaticUuid`;
+
+	let maxWidth = '';
+	let maxHeight = '';
+
+	if (amImageConfigurationEntry) {
+		const properties = amImageConfigurationEntry.properties;
+
+		const curMaxWidth = properties['max-width'];
+		const curMaxHeight = properties['max-height'];
+
+		if (curMaxWidth != 0) {
+			maxWidth = curMaxWidth;
+		}
+		if (curMaxHeight != 0) {
+			maxHeight = curMaxHeight;
+		}
+	}
+
+	const formik = useFormik({
+		initialValues: {
+			[descriptionId]: amImageConfigurationEntry
+				? amImageConfigurationEntry.description
+				: '',
+			[highResolutionId]: addHighResolution,
+			[maxHeightId]: maxHeight,
+			[maxWidthId]: maxWidth,
+			[nameId]: amImageConfigurationEntry
+				? amImageConfigurationEntry.name
+				: '',
+			[newUuidId]: configurationEntryUuid,
+			[`${namespace}uuid`]: configurationEntryUuid,
+		},
+		onSubmit: (values) => {
+			fetch(actionUrl, {
+				body: objectToFormData(values),
+				method: 'POST',
+			})
+				.then((res) => res.json())
+				.then(({message, success}) => {
+					if (success) {
+						const title = Liferay.Language.get('success');
+
+						openToast({
+							message,
+							title,
+							type: 'success',
+						});
+
+						Liferay.Util.navigate(redirect);
+					}
+					else {
+						setErrorMessage(message);
+						scrollToTop();
+					}
+				});
+		},
+		validate: (values) => {
+			const err = validate(
+				{
+					[nameId]: [required],
+					[newUuidId]: [alphanumeric],
+				},
+				values
+			);
+
+			if (!values[maxWidthId] && !values[maxHeightId]) {
+				err[maxWidthId] = Liferay.Language.get(
+					'at-least-one-value-is-required'
+				);
+
+				err[maxHeightId] = true;
```

Comment:
Will this `true` here have the effect of highlighting the field? What made you pick width as the one to have the message attached to it?

Reply:
Yes, the `true` highlights the field showing an error, but not a message. And the width shows the message just because is the first field. This was defined by UX some time ago, so we just kept that pattern. 

<img width="432" alt="Screenshot 2020-11-05 at 22 39 46" src="https://user-images.githubusercontent.com/8373764/98299538-23cccd00-1fb8-11eb-8a8f-e520ead9fdec.png">

Reply:
Thanks for the explanation. A ~picture~ screenshot is worth a thousand words in this case. 😂